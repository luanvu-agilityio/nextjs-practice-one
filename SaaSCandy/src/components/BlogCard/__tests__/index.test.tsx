import { render } from '@testing-library/react';
import { BlogCard } from '../index';
import Image from 'next/image';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: Record<string, unknown>) => (
    <Image src={''} alt={''} {...props} />
  ),
}));

jest.mock('next/link', () => ({
  __esModule: true,
  default: ({
    children,
    ...props
  }: { children: React.ReactNode } & Record<string, unknown>) => (
    <a {...props}>{children}</a>
  ),
}));

jest.mock('@/components/common', () => ({
  Heading: ({
    content,
    ...props
  }: { content: React.ReactNode } & Record<string, unknown>) => (
    <h3 {...props}>{content}</h3>
  ),
}));

describe('BlogCard', () => {
  const defaultProps = {
    slug: 'test-blog',
    title: 'Test Blog Title',
    date: '2024-01-01',
  };

  it('renders correctly with image', () => {
    const { container } = render(
      <BlogCard {...defaultProps} image='/test-image.jpg' />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly without image', () => {
    const { container } = render(<BlogCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
