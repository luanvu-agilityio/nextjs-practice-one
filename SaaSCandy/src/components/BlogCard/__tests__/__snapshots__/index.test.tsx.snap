// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BlogCard renders correctly with image 1`] = `
<div>
  <article
    class="group"
  >
    <a
      class="block rounded-lg overflow-hidden hover:shadow-md transition-shadow bg-white"
      href="/blog/test-blog"
    >
      <div
        class="w-full overflow-hidden"
      >
        <img
          alt="Test Blog Title"
          class="object-cover w-full h-48 sm:h-60 lg:h-71 group-hover:scale-105 transition-transform duration-300"
          height="284"
          src="/test-image.jpg"
          width="416"
        />
      </div>
      <div
        class="flex flex-col gap-3 sm:gap-4 py-6 sm:py-8 lg:py-10 px-4 sm:px-0"
      >
        <h3
          as="h3"
          class="text-primary font-medium text-lg sm:text-xl line-clamp-2"
        >
          Test Blog Title
        </h3>
        <div
          class="text-xs sm:text-sm text-gray-background"
        >
          2024-01-01
        </div>
      </div>
    </a>
  </article>
</div>
`;

exports[`BlogCard renders correctly without image 1`] = `
<div>
  <article
    class="group"
  >
    <a
      class="block rounded-lg overflow-hidden hover:shadow-md transition-shadow bg-white"
      href="/blog/test-blog"
    >
      <div
        class="w-full overflow-hidden"
      >
        <div
          class="w-full h-48 sm:h-60 lg:h-71 flex items-center justify-center text-gray-background bg-gray-100"
        >
          No image
        </div>
      </div>
      <div
        class="flex flex-col gap-3 sm:gap-4 py-6 sm:py-8 lg:py-10 px-4 sm:px-0"
      >
        <h3
          as="h3"
          class="text-primary font-medium text-lg sm:text-xl line-clamp-2"
        >
          Test Blog Title
        </h3>
        <div
          class="text-xs sm:text-sm text-gray-background"
        >
          2024-01-01
        </div>
      </div>
    </a>
  </article>
</div>
`;
